gazelle_haskell_modules/rule_generation.go

```diff
@@ -459,3 +500,44 @@ func shouldModularize(r *rule.Rule) bool {
 	}
 	return true
 }
+
+// Detect srcs specifications from comments attached to a rule.
+//
+// # gazelle_haskell_modules:srcs: src/ pirin/
+// ->
+// [src/ pirin/]
+func getSrcDirsFromRuleDirective(r *rule.Rule) ([]string, error) {
+	return getSrcDirsFromComments(r.Comments())
+}
+
+func getSrcDirsFromComments(cs []string) ([]string, error) {
+
+	srcsComment := ""
+	for _, comment := range cs {
+		if strings.Contains(comment, PRIVATE_FIND_MODULES_DIRECTIVE) {
+			if srcsComment != "" {
+				return nil, fmt.Errorf("found more than one srcs directive")
+			}
+			srcsComment = comment
+		}
+	}
+	if srcsComment == "" {
+		return []string{}, nil
+	}
+	stripped, err := stripDirective(srcsComment)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return strings.Fields(stripped), nil
+}
+
+func stripDirective(c string) (string, error) {
+	regex := regexp.MustCompile(fmt.Sprintf(`#\s*%s(.*)`, PRIVATE_FIND_MODULES_DIRECTIVE))
+	matches := regex.FindStringSubmatch(c)
+	if len(matches) != 2 {
+		return "", fmt.Errorf("didn't find leading directive")
+	}
+	return matches[1], nil
+}
```

Comment:
Compiling a regex for every comment looks like a bit of overkill. Perhaps the regex should be be made global. That said, I would expect to be easy to do without regexes in this case. 

`gazelle_haskell_modules` will produce an error when given a comment like `# asdf gazelle_haskell_modules:srcs: blah` because it doesn't match the regex. Shouldn't this comment be silently ignored instead?

Reply:
My previous implementation would instead detect that, which @YorikSar thinks is confusing. But I can imagine myself being confused by it getting silently ignored too. I'll mode the regex out though, good point.

Reply:
I think ignoring comments that aren't directives is what most people would expect. And directives are comments starting with the syntax `# gazelle:directive ...`. The former version of the code wasn't ignoring the comment, it was treating it as a directive, which wouldn't be what I would expect either.

Reply:
Done